src/main/java/homework_4/сustom_annotation/SimpleClassWithNameField.java

```diff
@@ -0,0 +1,25 @@
+package homework_4.сustom_annotation;
+
+import java.lang.reflect.Constructor;
+
+public class SimpleClassWithNameField {
+    private String name;
+
+    @CustomAnnotation("test name string")
+    public SimpleClassWithNameField(String name) {
+        Constructor<? extends SimpleClassWithNameField> constructor;
+        try {
+            constructor = this.getClass().getConstructor(String.class);
+            CustomAnnotation annotation = constructor.getAnnotation(CustomAnnotation.class);
+            if (annotation != null) this.name = annotation.value();
+            if (!name.isEmpty()) this.name = name;
```

Comment:
Opt: 
1) let's not use if without brackets, this approach is prone to errors
2) I think, the idea you try to implement would look better if 'if-else' or ternary operator will be used

Reply:
add brackets